neurots/generate/algorithms/abstractgrower.py

```diff
@@ -23,7 +23,7 @@ class AbstractAlgo:
 
     def __init__(self, input_data, params, start_point, context):
         """The TreeGrower Algorithm initialization."""
-        self.context = context
+        self.context = context if context is not None else {}
```

Comment:
In the existing functions we use statements like `if self.context is not None and ...` so you should make it consistent either by fixing all existing `if` statements or by fixing you new statements to handle the `None` case.

Reply:
I'm not sure how to fix in fact, as the condition after None assume a dict with a specific key, so if we have {} instead of None, the if will work the same, no?

Reply:
I didn't say it can't work, I just said it's not consistent :)
If you ensure the context can never be `None`, then you should remove all the `if self.context not is None ...`. But this argument is still a bit experimental and there is not clear contract associated to it, so it's a bit hard to know how to handle it. For now I think it's always supposed to be a `dict` with some arbitrary items, but it would be better to create a specific class for this context object. But that's for our successors :smile:

Reply:
ok, I removed the is None checks